Update Cancellation Survey and Expand Questions

dzver opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Part of #

## Proposed Changes

* Update the cancellation survey based on the work by @bohdankit07 on p58i-lsX-p2
* Introduce many new questions
* Introduce other options with textareas

## Why are these changes being made?

* Step one of compliance changes coming up next week

## Testing Instructions

* Turn on the store sandbox
* Switch locale to English
* Buy a plan
* Cancel it
* Switch locale to any non-English locale
* Ensure the old survey is still there and functioning

A sample of how I tested it:
https://github.com/user-attachments/assets/c6fdc9a8-4313-427a-a144-2804f2fe100d

Then I confirmed the expected structure with wpsh 38242-pb

```
    [why-cancel] => Array
        (
            [response] => otherDomain
            [text] => asdf
        )
```

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

dzver committed:
Add first two new questions

dzver committed:
Move technical issues up

dzver committed:
Move domain up

dzver committed:
Update domains

dzver committed:
Add wrong plan or site

dzver committed:
Add bad support and no longer need site

dzver committed:
Add textarea and the first new Other

dzver committed:
Add freetext other reasons to each option

dzver committed:
Consistency with grammar

dzver committed:
Disable the button on submit

Also shuffle the solutions

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17553403

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @dzver, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Too expensive`
- `The plan is too expensive for the features offered.`
- `Lack of customization features (e.g., colors, fonts, themes).`
- `The free plan is sufficient for my current needs.`
- `I found a competitor with better pricing/value.`
- `Budget constraints / Can no longer afford it.`
- `Something else related to price/value.`
- `Please tell us more about your price/value concern`
- `Too hard to use`
- `The platform/dashboard is too complicated or confusing to navigate.`
- `The website editor is difficult or unintuitive to use.`
- `It takes too much time to learn how to use the platform.`
- `Onboarding or tutorials were not helpful enough.`
- `Something else related to too hard to use.`
- `Please tell us more`
- `Cannot install desired plugins (e.g., from WordPress.org or third-party).`
- `Cannot upload custom themes.`
- `Limited design or customization options (e.g., layout, specific elements).`
- `Lacking specific functionality I need (e.g., e-commerce, specific integrations, advanced SEO tools).`
- `Core features I expected are only available on a much higher plan.`
- `Something else related to features/flexibility.`
- `Technical problems`
- `Site is slow or experiences performance issues.`
- `Encountered bugs, errors, or glitches.`
- `Problems migrating my existing site or content.`
- `Site was down or inaccessible.`
- `Something else related to technical problems.`
- `Problems with domain`
- `Trouble connecting or transferring my existing domain.`
- `Confused about how domains work with the WordPress.com plan.`
- `Didn’t get the domain name I expected / My domain is incorrect.`
- `Something else related to domains.`
- `Wrong plan or site`
- `I purchased this plan by mistake.`
- `I meant to upgrade a different website or account.`
- `The plan I chose didn’t match what I thought I was buying.`
- `Something else related to an accidental purchase.`
- `Bad support experience`
- `Support was unhelpful or slow to respond.`
- `Could not easily reach a human support agent.`
- `AI/automated support was not sufficient for my issue.`
- `Something else related to support.`
- `No longer need a site`
- `My project/business plans have changed.`
- `I was just exploring/testing the platform.`
- `I no longer need this website/service.`
- `Temporary cancellation, I might return.`
- `Something else related to changed needs.`

Thank you in advance!

bohdankit07 commented:
> do you mind sharing more about the tracking event request? Is there an issue or a p2 about it?
> 
> I don't mind fixing it here or in a separate PR, depending on how big of a change it is, but I'm not sure what the problem is. I searched the linear issue and the P2 post and couldn't find the problem. I was also not aware it existed.

I have shared this in Slack in the original request, but p58i-lR3-p2#comment-68832. However, since we are changing how the flow works now due to the FTC's Negative Rule, my recommendations are no longer fully applicable to how we should track the performance of upsells, and we likely need a tracking approach that allows us to connect cancellations with upsell clicks post-cancellation, and an auto-renewal's turning back on.

dzver committed:
Refactor the other reason

dzver committed:
Use the answer of the second question, if it exists

dzver committed:
We'll now the value of the second select, if it exists

when submitting a textarea

dzver committed:
Fix typescript problem

chriskmnds commented on client/components/marketing-survey/cancel-purchase-form/index.jsx:
```diff
@@ -256,9 +262,14 @@ class CancelPurchaseForm extends Component {
 				isSubmitting: true,
 			} );
 
+			const useSubOption = this.state.questionOneDetails && this.state.questionOneText;
```
NIT - just seeing `useSubOption ` here -`useFoo` being the naming convention for hooks. I'd maybe pick a different name on that basis, but also fine since it doesn't confuse the linter

This will also reference the result of `this.state.questionOneDetails && this.state.questionOneText`. If we were to be pedantic, maybe consider making it a boolean.

chriskmnds commented on client/components/marketing-survey/cancel-purchase-form/index.jsx:
```diff
@@ -139,6 +139,7 @@ class CancelPurchaseForm extends Component {
 			atomicRevertCheckOne: false,
 			atomicRevertCheckTwo: false,
 			purchaseIsAlreadyExtended: false,
+			nextAdventureValidation: true,
```
NIT - "validation" doesn't exactly point to a boolean (?), more like a concept/object. Perhaps I'd name it as `isNextAdventureValid`, which may be more idiomatic.

chriskmnds commented on client/components/marketing-survey/cancel-purchase-form/step-components/feedback-step.tsx:
```diff
@@ -8,26 +8,37 @@ import { toSelectOption } from '../to-select-options';
 	const translate = useTranslate();
 	const [ value, setValue ] = useState( '' );
 	const [ details, setDetails ] = useState( '' );
+	const [ textAreaValue, setTextAreaValue ] = useState( '' );
```
Any chance we can give this a more semantic name? Something with "feedback" perhaps?

chriskmnds commented on client/components/marketing-survey/cancel-purchase-form/step-components/feedback-step.tsx:
```diff
@@ -37,6 +48,7 @@ function CancellationReason( { purchase, reasonCodes, ...props }: CancellationRe
 					options={ reasons.map( toSelectOption ) }
 					onChange={ ( val ) => {
 						onDetailsChange( '' );
+						setTextAreaValue( '' );
```
This is a little ambiguous how we'd call `setTextAreaValue` here and in `onDetailsChange` above. It looks unnecessary to do it in both places.

chriskmnds commented on client/components/marketing-survey/cancel-purchase-form/step-components/feedback-step.tsx:
```diff
@@ -8,26 +8,37 @@ import { toSelectOption } from '../to-select-options';
 
 	const onDetailsChange = ( val: string ) => {
 		setDetails( val );
+		setTextAreaValue( '' );
```
I think we could just call `onTextAreaChange` here and avoid repeating the same logic

chriskmnds commented on client/components/marketing-survey/cancel-purchase-form/cancellation-reasons.ts:
```diff
@@ -142,93 +166,240 @@ export const CANCELLATION_REASONS: CancellationReason[] = [
+				value: 'domainIncorrect',
 				get label() {
-					return translate( 'Problem connecting my domain' );
+					return translate( 'Didn’t get the domain name I expected / My domain is incorrect.' );
```
Just noting the backtick in `Didn’t`
Perhaps wrap the string in double quotes and use a vertical apostrophe instead? I think that's the convention.

chriskmnds commented on client/components/marketing-survey/cancel-purchase-form/cancellation-reasons.ts:
```diff
@@ -142,93 +166,240 @@ export const CANCELLATION_REASONS: CancellationReason[] = [
+			{
+				value: 'noMatch',
+				get label() {
+					return translate( 'The plan I chose didn’t match what I thought I was buying.' );
```
Use of backtick - `didn’t`

chriskmnds reviewed (commented):
Heya. Checked the code and left some comments. I don't think I checked everything, though.

Looks good overall. Some NIT comments

dzver committed:
Rename variable per code review request

dzver committed:
Rename variable per request

use is reserved for hooks

dzver committed:
Rename variables based on feedback

dzver committed:
Remove duplicate call

dzver commented:
I addressed all comments except the single quote, I'll leave that unaddressed for now.

My personal preference is to use ' rather than the curly single quote because it's supported by Cursor. However, this entire area of code uses the curly single quote and changing that feels like a subject to a separate change and something to discuss with i18n folks.
